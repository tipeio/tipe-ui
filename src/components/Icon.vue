<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
