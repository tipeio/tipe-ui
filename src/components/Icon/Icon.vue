<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
