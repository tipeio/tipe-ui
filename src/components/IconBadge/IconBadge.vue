<template>
  <div class="card">
    <tipe-icon
      :icon="icon"
      class="icon"
      height="1.25rem"
      width="1.25rem"
    />
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../Icon'

export default {
  name: 'IconBadge',
  components: {
    TipeIcon
  },
  props: {
    icon: vueTypes.string.isRequired
  }
}
</script>

<style scoped lang="postcss">
.card {
  width: 3.4375rem;
  height: 3.4375rem;
  border-radius: 4px;
  background-color: var(--purple-transparent);
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
