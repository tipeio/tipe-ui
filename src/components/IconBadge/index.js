export { default } from './IconBadge.vue'
