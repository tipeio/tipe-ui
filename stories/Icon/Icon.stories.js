import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
