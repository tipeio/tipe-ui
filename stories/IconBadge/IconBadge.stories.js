import { storiesOf } from '@storybook/vue'

import IconBadge from '../../src/components/IconBadge'

storiesOf('IconBadge', module).add('icon badge', () => ({
  components: { IconBadge },
  template: '<icon-badge  icon="file" />'
}))
