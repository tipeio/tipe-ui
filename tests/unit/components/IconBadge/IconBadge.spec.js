import { shallowMount } from '@vue/test-utils'
import IconBadge from '@/components/IconBadge'
import Icon from '@/components/Icon'

describe('IconBadge.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(IconBadge, {
      propsData: {
        icon: 'rocket'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
  it('should render with the card class', () => {
    const wrapper = shallowMount(IconBadge, {
      propsData: {
        icon: 'rocket'
      }
    })
    expect(wrapper.classes()).toContain('card')
  })
  it('should pass the icon prop', () => {
    const wrapper = shallowMount(IconBadge, {
      propsData: {
        icon: 'rocket'
      }
    })
    expect(wrapper.props().icon).toBe('rocket')
  })
  it('should contain the icon component', () => {
    const wrapper = shallowMount(IconBadge, {
      propsData: {
        icon: 'rocket'
      }
    })
    expect(wrapper.find(Icon).exists()).toBe(true)
  })
})
