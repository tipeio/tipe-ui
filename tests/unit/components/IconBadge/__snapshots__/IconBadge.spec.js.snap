// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBadge.vue renders 1`] = `
<div class="card">
  <tipeicon-stub icon="rocket" height="1.25rem" width="1.25rem" color="#000" __iconloader__="function _default(icon) {
  return require(&quot;../../icons/&quot;.concat(icon, &quot;.svg&quot;)).default;
}" class="icon"></tipeicon-stub>
</div>
`;
